sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java

```diff
@@ -871,4 +871,18 @@ public void testJoinBranchReorderingOnMultipleTables() throws Exception {
             is("0| 0| 0\n" +
                "1| 1| 1\n"));
     }
+
+    @Test
+    public void testInnerJoinWithPushDownOptimizations() {
+        execute("CREATE TABLE t1 (id INTEGER)");
+        execute("CREATE TABLE t2 (id INTEGER, name STRING, id_t1 INTEGER)");
+
+        execute("INSERT INTO t1 (id) VALUES (1), (2)");
+        execute("INSERT INTO t2 (id, name, id_t1) VALUES (1, 'A', 1), (2, 'B', 2), (3, 'C', 2)");
+        execute("REFRESH TABLE t1, t2");
+
+        execute("SELECT t1.id, t2.id FROM t2 INNER JOIN t1 ON t1.id = t2.id_t1 ORDER BY lower(t2.name)");
+        execute("SELECT t1.id, t2.id, t2.name FROM t2 INNER JOIN t1 ON t1.id = t2.id_t1 ORDER BY lower(t2.name)");
+        execute("SELECT t1.id, t2.id, lower(t2.name) FROM t2 INNER JOIN t1 ON t1.id = t2.id_t1 ORDER BY lower(t2.name)");
```

Comment:
as there is no assertion inside the test, could you please write a comment, why?